x-pack/plugins/case/server/client/cases/update.ts

```diff
@@ -95,133 +202,158 @@ export const update = ({
+    requests: updateFilterCases,
+    caseService,
+    client: savedObjectsClient,
+  });
 
-    for (const theCase of [
-      ...casesWithSyncSettingChangedToOn,
-      ...casesWithStatusChangedAndSynced,
-    ]) {
-      const currentCase = myCases.saved_objects.find((c) => c.id === theCase.id);
-      const totalComments = await caseService.getAllCaseComments({
-        client: savedObjectsClient,
-        caseId: theCase.id,
-        options: {
-          fields: [],
-          filter: 'cases-comments.attributes.type: alert',
-          page: 1,
-          perPage: 1,
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  const { username, full_name, email } = await caseService.getUser({ request });
+  const updatedDt = new Date().toISOString();
+  const updatedCases = await caseService.patchCases({
+    client: savedObjectsClient,
+    cases: updateFilterCases.map((thisCase) => {
+      const { id: caseId, version, ...updateCaseAttributes } = thisCase;
+      let closedInfo = {};
+      if (updateCaseAttributes.status && updateCaseAttributes.status === CaseStatuses.closed) {
+        closedInfo = {
+          closed_at: updatedDt,
+          closed_by: { email, full_name, username },
+        };
+      } else if (
+        updateCaseAttributes.status &&
+        (updateCaseAttributes.status === CaseStatuses.open ||
+          updateCaseAttributes.status === CaseStatuses['in-progress'])
+      ) {
+        closedInfo = {
+          closed_at: null,
+          closed_by: null,
+        };
+      }
+      return {
+        caseId,
+        updatedAttributes: {
+          ...updateCaseAttributes,
+          ...closedInfo,
+          updated_at: updatedDt,
+          updated_by: { email, full_name, username },
         },
-      });
+        version,
+      };
+    }),
+  });
 
-      const caseComments = (await caseService.getAllCaseComments({
-        client: savedObjectsClient,
-        caseId: theCase.id,
-        options: {
-          fields: [],
-          filter: 'cases-comments.attributes.type: alert',
-          page: 1,
-          perPage: totalComments.total,
-        },
-        // The filter guarantees that the comments will be of type alert
-      })) as SavedObjectsFindResponse<{ alertId: string }>;
-
-      const commentIds = caseComments.saved_objects.map(({ attributes: { alertId } }) => alertId);
-      if (commentIds.length > 0) {
-        caseClient.updateAlertsStatus({
-          ids: commentIds,
-          // Either there is a status update or the syncAlerts got turned on.
-          status: theCase.status ?? currentCase?.attributes.status ?? CaseStatuses.open,
-        });
-      }
-    }
+  // If a status update occurred and the case is synced then we need to update all alerts' status
+  // attached to the case to the new status.
+  const casesWithStatusChangedAndSynced = updateFilterCases.filter((caseToUpdate) => {
+    const currentCase = myCases.saved_objects.find((c) => c.id === caseToUpdate.id);
```

Comment:
We can use your beautiful `casesMap` here to avoid the search.

Reply:
Yeah, let's do that in a follow up PR. This all looks like this changed but I just unindented it.